sklearn/dummy.py

```diff
@@ -143,13 +143,17 @@ def fit(self, X, y, sample_weight=None):
          self.n_classes_,
          self.class_prior_) = class_distribution(y, sample_weight)
 
-        if (self.strategy == "constant" and
-                any(constant[k] not in self.classes_[k]
-                    for k in range(self.n_outputs_))):
-            # Checking in case of constant strategy if the constant
-            # provided by the user is in y.
-            raise ValueError("The constant target value must be "
-                             "present in training data")
+        if self.strategy == "constant":
+            for k in range(self.n_outputs_):
+                if constant[k] not in self.classes_[k]:
+                    # Checking in case of constant strategy if the constant
+                    # provided by the user is in y.
+                    raise ValueError(
+                        "The constant target value must be present "
+                        "in the training data. "
+                        "You provided constant=%r. "
+                        "Possible values are: %r." % (self.constant,
+                                                      list(self.classes_[k])))
```

Comment:
why ńot using formating? 

```py
_ERR_MSG = (
"The constant ... "
" Possible values (from training data) are: {possible_values}"
" You provided {provided}."
).format(possible_values=list(self.classes_[k]), provided=self.constant)
```

and then just call `raise ValueError(_ERR_MSG)`

Reply:
I used the % approach to keep consistency with the rest of the script. But I will define the error message separately to improve readability.

Reply:
I think this is not quite usual in scikit-learn:

```python
err_msg = ("The constant target value must be present in the training target."
           "Got {} while expecting one of the following: {}
           .format(self.constant, self.classes_[k])".)
```

Reply:
@martinagvilas if `format` is used nowhere else in `sklearn/dummy.py` I would then use `%` style. If format is already there then I would use format.

It is true that `err_msg` is better than `_MSG`.